extensions/resource-deployment/src/services/kubeService.ts

```diff
@@ -29,28 +29,31 @@ export class KubeService implements IKubeService {
 }
 
 export function getKubeConfigClusterContexts(configFile: string): Promise<KubeClusterContext[]> {
+	let contexts: KubeClusterContext[];
 	return fs.promises.access(configFile).catch((error) => {
 		if (error && error.code === 'ENOENT') {
-			return [];
+			contexts = [];
 		}
 		else {
 			throw error;
 		}
 	}).then(() => {
-		const config = yamljs.load(configFile);
-		const rawContexts = <any[]>config['contexts'];
-		const currentContext = <string>config['current-context'];
-		const contexts: KubeClusterContext[] = [];
-		if (currentContext && rawContexts && rawContexts.length > 0) {
-			rawContexts.forEach(rawContext => {
-				const name = <string>rawContext['name'];
-				if (name) {
-					contexts.push({
-						name: name,
-						isCurrentContext: name === currentContext
-					});
-				}
-			});
+		if (contexts === undefined) {
```

Comment:
Instead of this why not just make this async and use try/catch? Then this logic would just be inside the try and so the undefined check would be unnecessary.

Reply:
done